src/providers/java.rs

```diff
@@ -29,15 +31,17 @@ impl Provider for JavaProvider {
     }
 
     fn get_build_plan(&self, app: &App, _env: &Environment) -> Result<Option<BuildPlan>> {
-        let mut setup = Phase::setup(Some(vec![Pkg::new("jdk")]));
-
+        let mut setup: Phase;
         let build = if self.is_using_gradle(app) {
-            setup.add_nix_pkgs(vec![Pkg::new("gradle")]);
+            let pkgs = self.get_jdk_and_gradle_pkgs(app)?;
+            setup = Phase::setup(Some(pkgs));
+
             let gradle_exe = self.get_gradle_exe(app);
             let mut build = Phase::build(Some(format!("{} build", gradle_exe)));
-            build.add_cache_directory(".gradle");
+            build.add_cache_directory("~/.gradle");
```

Comment:
Did you confirm this works? I'm not sure the `~` works in cache locations.

Reply:
here https://github.com/railwayapp/nixpacks/blob/main/src/nixpacks/builder/docker/utils.rs#L11

the `~` is being replaced with `root`
Will push now a more clear path to avoid this confusion 


This screenshot is from inside the container, as you can see the cache exists at `/root/.gradle`
<img width="609" alt="image" src="https://user-images.githubusercontent.com/781863/188928547-69346d4b-c9cf-48d7-8870-322c4393ac36.png">

Reply:
Ok, seems that we have both `/root/.gradle` & `.gradle` in the container 
After more investigation, the active one is `.gradle` I've changed it back now